src/newgrf_badge.h

```diff
@@ -24,7 +24,7 @@ class Badge {
 	BadgeFlags flags = {}; ///< Display flags
 	StringID name = 0; ///< Short name.
 	GrfSpecFeatures features{}; ///< Bitmask of which features use this badge.
-	VariableGRFFileProps grf_prop; ///< Sprite information.
+	FixedGRFFileProps<GrfSpecFeature, GrfSpecFeature::GSF_FAKE_END> grf_prop; ///< Sprite information.
```

Comment:
I used `VariableGRFFileProps` here to limit the size (with the fixed array, each badge will grow from 88 to 248 bytes). But possibly per-feature graphics is a misdesign anyway.

Reply:
Fine, more templates then :)

Reply:
Yeah, there's a reason I used a generic type...

Reply:
It's only a small template. And once we switch to C++23 and `std::flat_map` all of it comes from the standard headers anyway.